modape/scripts/modis_collect.py

```diff
@@ -119,18 +153,31 @@ def cli(src_dir: str,
         group_files = [str(x) for x in hdf_files if group_pattern.match(x.name)]
         group_files.sort()
 
-        parameters = dict(
-            targetdir=targetdir,
-            files=group_files,
-            interleave=interleave,
-            group_id=group.replace('M.', 'MX'),
-            compression=compression,
-            vam_product_code=vam_code,
-            last_collected=last_collected,
-            force=force,
-        )
+        _raw_h5 = ModisRawH5(files=group_files,
+                            targetdir=targetdir,
+                            vam_product_code=vam_code,
+                            interleave=interleave)
+
+        if last_collected is not None:
+
+            if not _raw_h5.exists:
+                raise ValueError("Output H5 %s does not exist! Can't check last-collected!" % _raw_h5.filename)
 
-        processing_dict.update({group: parameters})
+            last_collected_infile = _raw_h5.last_collected
+
+            if not last_collected_infile:
+                raise ValueError(f"No last_collected recorded in {_raw_h5.filename}")
+
+            if not last_collected == last_collected_infile:
+                raise ValueError(f"Last collected date in file is {last_collected_infile} not {last_collected}")
+
+        to_process.update({
+            group: {
+                "raw_h5": _raw_h5,
+                "compression": compression,
+                "force": force,
+            }
+        })
```

Comment:
refactored the `_worker` function so `ModisRawH5` instances can be created ahead of processing, allowing the check on `last_collected` before anything is processed